src/main/java/org/gridsuite/explore/server/services/StudyService.java

```diff
@@ -37,14 +38,17 @@ public void setStudyServerBaseUri(String studyServerBaseUri) {
     }
 
     public void insertStudyWithExistingCaseFile(UUID studyUuid, String userId, UUID caseUuid, String caseFormat,
-            Map<String, Object> importParams, Boolean duplicateCase) {
-        String path = UriComponentsBuilder.fromPath(DELIMITER + STUDY_SERVER_API_VERSION +
+            Map<String, Object> importParams, Boolean duplicateCase, String firstRootNetworkName) {
+        var uriComponentsBuilder = UriComponentsBuilder.fromPath(DELIMITER + STUDY_SERVER_API_VERSION +
                 "/studies/cases/{caseUuid}")
                 .queryParam("studyUuid", studyUuid)
                 .queryParam("duplicateCase", duplicateCase)
-                .queryParam("caseFormat", caseFormat)
-                .buildAndExpand(caseUuid)
-                .toUriString();
+                .queryParam("caseFormat", caseFormat);
+
+        if (!StringUtils.isBlank(firstRootNetworkName)) {
+            uriComponentsBuilder.queryParam("firstRootNetworkName", firstRootNetworkName);
```

Comment:
you will need to encode this value in the URI, because it coud have spaces or other special characters

Reply:
well it seems to work after test then ok